activerecord/lib/active_record/connection_adapters/postgresql_adapter.rb

```diff
@@ -416,18 +416,24 @@ def supports_lazy_transactions?
         true
       end
 
+      def generate_advisory_lock_id
+        db_name_hash = Zlib.crc32(self.current_database)
+        schema_name_hash = Zlib.crc32(self.current_schema)
+        [db_name_hash, schema_name_hash].pack("L*").unpack("l*")
+      end
+
       def get_advisory_lock(lock_id) # :nodoc:
-        unless lock_id.is_a?(Integer) && lock_id.bit_length <= 63
-          raise(ArgumentError, "PostgreSQL requires advisory lock ids to be a signed 64 bit integer")
+        unless lock_id.is_a?(Array) && lock_id.all? { |id| id.is_a?(Integer) && id.bit_length <= 31 }
+          raise(ArgumentError, "PostgreSQL advisory lock should be an array of two signed 32 bit integers")
         end
-        query_value("SELECT pg_try_advisory_lock(#{lock_id})")
+        query_value("SELECT pg_try_advisory_lock(#{lock_id[0]}, #{lock_id[1]})")
       end
 
       def release_advisory_lock(lock_id) # :nodoc:
-        unless lock_id.is_a?(Integer) && lock_id.bit_length <= 63
-          raise(ArgumentError, "PostgreSQL requires advisory lock ids to be a signed 64 bit integer")
+        unless lock_id.is_a?(Array) && lock_id.all? { |id| id.is_a?(Integer) && id.bit_length <= 31 }
+          raise(ArgumentError, "PostgreSQL advisory lock should be an array of two signed 32 bit integers")
```

Comment:
Same thing here, also, it looks like this code is the exact same as in `get_advisory_lock`, could it be extracted into a method:

```ruby
ADVISORY_LOCK_ERROR_MSG = "PostgreSQL advisory lock should be an array of two signed 32 bit integers".freeze

def get_advisory_lock(lock_id)
  raise(ArgumentError, ADVISORY_LOCK_ERROR_MSG) unless valid_lock_id?(lock_id)
 
  # ...
end

def release_advisory_lock(lock_id)
  raise(ArgumentError, ADVISORY_LOCK_ERROR_MSG) unless valid_lock_id?(lock_id)
 
  # ...
end

def valid_lock_id?(lock_id)
  return false unless lock_id.is_a?(Array)

  lock_id.count == 2 && lock_id.all? { |id| id.is_a?(Integer) && id.bit_length <= 31 }
end
```

Reply:
@ryanswood - Want to take a look at this? This would help unlock parallel migrations.

Reply:
These changes would be helpful. 

@leshik thoughts on the suggested changes?

Reply:
@ryanswood no problem, updated as suggested.